src/Document/Metadata/CacheAwareMetadataFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Document\Metadata;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Setono\SyliusMeilisearchPlugin\Document\Document;
+use Webmozart\Assert\Assert;
+
+final class CacheAwareMetadataFactory implements MetadataFactoryInterface
+{
+    /**
+     * The loaded metadata, indexed by class name
+     *
+     * @var array<class-string<Document>, MetadataInterface>
+     */
+    private array $loadedClasses = [];
+
+    public function __construct(
+        private readonly MetadataFactoryInterface $baseMetadataFactory,
+        private readonly CacheItemPoolInterface $cache,
+        private readonly bool $cacheEnabled = false,
```

Comment:
I don't think this service should be created when cache is false, i.e. it should be a conditional service based on the configuration value of the cache parameter

Reply:
👍